src/Compilers/CSharp/Test/Syntax/Parsing/MemberDeclarationParsingTests.cs

```diff
@@ -1915,17 +1934,30 @@ public void RequiredModifier_LocalNamedRequired_TopLevelStatements(CSharpParseOp
         {
-            var error =
-                // (1,12): error CS8773: Feature 'static abstract members in interfaces' is not available in C# 9.0. Please use language version 11.0 or greater.
-                // public int N.I.operator +(int x, int y) => x + y;
-                Diagnostic(ErrorCode.ERR_FeatureNotAvailableInVersion9, "N.I.").WithArguments("static abstract members in interfaces", "11.0").WithLocation(1, 12);
+            var text = "public int N.I.operator +(int x, int y) => x + y;";
+            var classWithText = $"class C {{ {text} }}";
+            CreateCompilation(classWithText, parseOptions: TestOptions.Regular9).VerifyDiagnostics(
+                // (1,22): error CS0246: The type or namespace name 'N' could not be found (are you missing a using directive or an assembly reference?)
+                // class C { public int N.I.operator +(int x, int y) => x + y; }
+                Diagnostic(ErrorCode.ERR_SingleTypeNameNotFound, "N").WithArguments("N").WithLocation(1, 22),
+                // (1,22): error CS8773: Feature 'static abstract members in interfaces' is not available in C# 9.0. Please use language version 11.0 or greater.
+                // class C { public int N.I.operator +(int x, int y) => x + y; }
+                Diagnostic(ErrorCode.ERR_FeatureNotAvailableInVersion9, "N.I.").WithArguments("static abstract members in interfaces", "11.0").WithLocation(1, 22),
+                // (1,22): error CS0538: 'N.I' in explicit interface declaration is not an interface
+                // class C { public int N.I.operator +(int x, int y) => x + y; }
+                Diagnostic(ErrorCode.ERR_ExplicitInterfaceImplementationNotInterface, "N.I").WithArguments("N.I").WithLocation(1, 22),
+                // (1,35): error CS0106: The modifier 'public' is not valid for this item
+                // class C { public int N.I.operator +(int x, int y) => x + y; }
+                Diagnostic(ErrorCode.ERR_BadMemberFlag, "+").WithArguments("public").WithLocation(1, 35),
+                // (1,35): error CS8930: Explicit implementation of a user-defined operator 'C.operator +(int, int)' must be declared static
+                // class C { public int N.I.operator +(int x, int y) => x + y; }
+                Diagnostic(ErrorCode.ERR_ExplicitImplementationOfOperatorsMustBeStatic, "+").WithArguments("C.operator +(int, int)").WithLocation(1, 35));
 
             foreach (var options in new[] { TestOptions.Script, TestOptions.Regular })
             {
                 foreach (var version in new[] { LanguageVersion.CSharp9, LanguageVersion.Preview })
                 {
-                    UsingDeclaration("public int N.I.operator +(int x, int y) => x + y;", options: options.WithLanguageVersion(version),
-                        version == LanguageVersion.CSharp9 ? new[] { error } : new DiagnosticDescription[] { });
+                    UsingDeclaration(text, options: options.WithLanguageVersion(version));
```

Comment:
these tests demonstrate the value in this change.  With the parser changes, you now get the same tree regardless of what langversion you parse with.  This is the improvement that will allow the IDE to share trees across project targets with different versions.  Differences in diagnostics should be in the binding layer only.